lib/random.gi

```diff
@@ -29,9 +29,11 @@ InstallMethod(Reset, [IsRandomSource], function(rs)
   return Reset(rs, 1);
 end);
 
-# Generic fallback methods, such that it is sufficient to install Random for 
-# lists or for pairs of integers.
-InstallMethod(Random, [IsRandomSource, IsInt, IsInt], function(rs, a, b)
+DeclareGlobalFunction("RANDOM_LARGE_RANGE");
+
+# Generic method to build an arbitary random number generator from a generator
+# which can generate numbers at least in the range [0..2^10].
+InstallGlobalFunction("RANDOM_LARGE_RANGE", function(gen, a, b)
```

Comment:
Why 2^10 and not e.g. 2^28? How about making the exponent "10" an argument to the function, for optimal performance?
(I am also OK with leaving this as is, though)

Reply:
This just came from pulling out an old function fairly unchanged. When I did this I had hoped to not change the random numbers produced (that turned out to be impossible for other reasons).